docs/articles/documentation/how-it-works/README.md

```diff
@@ -3,151 +3,40 @@ layout: docs
 ---
-# How It Works
+# How Does TestCafe Work?
 
-Before TestCafe, end-to-end web testing tools that worked with real browsers relied on driving these browsers from the outside to emulate user actions. This approach, however, had serious drawbacks.
+The automation engine behind TestCafe takes over browsers and the web pages they display to simulate user activity.
 
-* **Dependency on third-party tools.** This required you to install, configure, and maintain additional drivers, SDKs and/or libraries in every test environment.
-* **Limited browser support.** The range of available browsers and platforms was restricted to those supported by the driver.
-* **Boilerplate code.** These testing tools functioned with the same browser instance during the entire test run, so the tested web app state (cookies, storages, etc.) persisted from test to test. As a result, end-to-end tests contain a large amount of boilerplate code that clears the state to avoid interference between tests.
-* **Low automation.** Legacy tools were unable to interact directly with the tested page. Although the fact that the user’s test code did not intervene in the app code was an advantage, the testing tool itself could benefit from operating inside the page. This would enable the tool to automate many routine tasks.
+## The Server-Client Architecture
 
-## TestCafe is Different
+The server-client architecture of TestCafe makes it capable of executing both system-level and in-browser code.
 
-The idea that defined TestCafe architecture was that you don't really need an external driver to run end-to-end tests in the browser. Instead, the page can run all the test scripts that emulate user actions. This would allow for true cross-browser and cross-platform tests that run on any device that has a modern browser. This vision eventually developed into the following design concepts.
+* TestCafe uses [high-level system APIs](https://github.com/DevExpress/testcafe-browser-tools) to launch and manage browsers. This is necessary to achieve full control over test execution.
+* TestCafe tests are full-fledged Node.js scripts. They can launch services and applications, read and write system files, make use of your favourite libraries.
+* At the same time, TestCafe is capable of performing in-browser actions via client-side automation scripts. This is how TestCafe handles asynchronous events, simulates user activity, and executes custom (user-defined) JavaScript.
 
-### Client-server architecture
+Our experience shows that this hybrid approach gives users the best of both worlds. Early in the product’s life, TestCafe ran entirely in the browser. This caused a number of issues. The tests’ capabilities were limited, the application often interfered with page execution, and the sub-optimal logic separation caused browser crashes. Since then, we decoupled the engine, exposed the underlying Node.js API, and implemented a test runner that executes test scripts server-side.
```

Comment:
> Early in the product’s life, TestCafe ran entirely in the browser.

This line got me thinking 'which product?' for a second.

Perhaps we could do it shorter, like this:

> Earlier versions of TestCafe ran entirely in the browser.

Reply:
Great suggestion!

Reply:
On a second thought, '_earlier versions_' can be interpreted as 'all versions of the product up until now', which would be clearly false. Let me think